extensions/vscode/src/daemon/dart-frog-application-registry.ts

```diff
@@ -0,0 +1,277 @@
+import {
+  DaemonEvent,
+  DaemonRequest,
+  DartFrogApplication,
+  DartFrogDaemon as DartFrogDaemon,
+  DartFrogDaemonEventEmitterTypes,
+  isApplicationExitDaemonEvent,
+  isApplicationStartingDaemonEvent,
+  isLoggerInfoDaemonEvent,
+  isProgressCompleteDaemonEvent,
+  isStartDaemonRequest,
+} from ".";
+import { EventEmitter } from "events";
+
+/**
+ * The prefix of the message that is sent by the Dart Frog Daemon when the Dart
+ * VM service is listening.
+ *
+ * @example
+ * "The Dart VM service is listening on http://127.0.0.1:8181/fQBcSu3OOc8=/"
+ */
+const vmServiceUriMessagePrefix = "The Dart VM service is listening on ";
+
+/**
+ * The prefix of the message that is sent by the Dart Frog Daemon when the
+ * application is starting.
+ *
+ * @example
+ * "Running on \u001b]8;;http://localhost:8080\u001b\\http://localhost:8080\u001b]8;;\u001b\\"
```

Comment:
Do we need these escaping? They make it so much harder to read 😅

Reply:
This is the raw message as it comes from the daemon stdout.